[13.0][MIG] hr_employee_document

newtratip opened the pull request:
google_drive_link field is not in employee form, I removed it.

This PR is standard migration. Ready for review.

alexey-pelykh committed:
[12.0][ADD] hr_employee_document: various documents attached to the employee profile

oca-travis committed:
[UPD] Update hr_employee_document.pot

OCA-git-bot committed:
[ADD] icon.png

Commit:
Added translation using Weblate (French)

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (4 of 4 strings)

Translation: hr-12.0/hr-12.0-hr_employee_document
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_employee_document/fr/

OCA-git-bot committed:
[UPD] README.rst

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (4 of 4 strings)

Translation: hr-12.0/hr-12.0-hr_employee_document
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_employee_document/hr/

astirpe commented on hr_employee_document/__manifest__.py:
```diff
@@ -0,0 +1,16 @@
+    "installable": True,
+    "application": False,
+    "summary": "Documents attached to the employee profile",
+    "depends": ["hr", "attachment_indexation"],
```
Dependency with `attachment_indexation` is not necessary: with this module the attachments are already accessible by smart button.

newtratip replied:
Edited.

astirpe reviewed (changes requested):
Works fine!

alexey-pelykh commented on hr_employee_document/__manifest__.py:
```diff
@@ -2,22 +2,15 @@
-    'data': [
-        'views/hr_employee.xml',
-    ],
+    "name": "HR Employee Document",
```
Shouldn't this auto-format change be a separate commit?

alexey-pelykh commented on hr_employee_document/views/hr_employee_views.xml:
```diff
@@ -11,7 +11,6 @@
         <field name="inherit_id" ref="hr.view_employee_form"/>
         <field name="groups_id" eval="[(6, 0, [ref('hr.group_hr_user') ])]"/>
         <field name="arch" type="xml">
-            <field name="google_drive_link" position="replace"/>
```
This is part of migration or improvement?

astirpe replied:
Part of migration

newtratip replied:
Yes, Migration

alexey-pelykh replied:
That field is gone from Odoo?

newtratip replied:
@alexey-pelykh I can not see `google_drive_link` field in employee view.
![Selection_459](https://user-images.githubusercontent.com/24691983/97609749-9123af80-1a46-11eb-9886-42b6b87f3dfd.png)

alexey-pelykh commented:
Please tidy up the commit history, merge commit shouldn't even be there

robinkeunen committed:
[12.0][FIX] hr_employee_document: google_drive_link invisible

Replacing google drive link in view by nothing, that is deleting,
breaks the view inheritance.
Making it invisible instead.

alexey-pelykh commented:
Please also take a look at #799 as well

OCA-git-bot committed:
hr_employee_document 12.0.1.0.1

alexey-pelykh committed:
[IMP] hr_employee_document: add attachment preview

alexey-pelykh commented:
And cherry-pick #802

OCA-git-bot committed:
hr_employee_document 12.0.1.1.0

alexey-pelykh commented:
@Saran440 I think those cherry-picks should go before the migration and formatting commits

Saran440 commented:
@alexey-pelykh Okay, We will fix it. Thank you for your advise.
@newtratip Would you like to do this?

rafaelbn commented:
Hello @newtratip are you going to conitnue with this PR? Thank you!

newtratip committed:
[IMP] hr_employee_document: black, isort, prettier

newtratip commented:
> And cherry-pick #802

> @Saran440 I think those cherry-picks should go before the migration and formatting commits

@alexey-pelykh Done

alexey-pelykh commented:
Please follow https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0#technical-method-to-migrate-a-module-from-120-to-130-branch in splitting formatting and actual migration

newtratip committed:
[MIG] hr_employee_document: Migration to 13.0

newtratip commented:
@alexey-pelykh done, please check again.

newtratip commented:
@rafaelbn Yes, I continue with this PR. Can you help review please ?

alex-guardado commented on hr_employee_document/i18n/fr.po:
```diff
@@ -0,0 +1,38 @@
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: Odoo Server 12.0\n"
```
I don't know if this could cause a translation error. Maybe new translations should be made or the file should be changed to a more updated one. As I said i don't know if it's an error or it doesn't matter. :)

newtratip replied:
I don't sure translation file will be updated by oca-travis ? This below module just merged and oca-travis update automatically.
![Selection_480](https://user-images.githubusercontent.com/24691983/98136291-5d87d000-1ef3-11eb-9318-62cc0da14cd2.png)

Do you know about this ? @kittiu @Saran440

Saran440 replied:
@alex-guardado I think it doesn't matter. module will translate from .pot and bot will auto generate after merged.

alex-guardado replied:
Okey, thank you for your answers. I didn't know how it worked exactly. Thank you :)

astirpe reviewed (approved):
Code review only. LGTM.

Saran440 reviewed (approved):
Functional Test :+1:

alexey-pelykh commented:
/ocabot merge nobump
